test/jdk/javax/swing/SwingWorker/TestDoneBeforeDoInBackground.java

```diff
@@ -88,14 +88,22 @@ public void propertyChange(PropertyChangeEvent evt) {
                                "PropertyChangeListeners called with " +
                                "state STARTED after doInBackground is invoked");
                     }
+                    // Ensure the STARTED state is notified
+                    // before doInBackground started running
+                    if (!doInBackground &&
+                         worker.getState() == SwingWorker.StateValue.STARTED) {
```

Comment:
This condition doesn't guarantee that `STARTED` state is set before `doInBackground` starts running. The boolean flag is set to `true` only in the end of the method.